src/reader.rs

```diff
@@ -715,10 +720,24 @@ impl<B: BufRead> Reader<B> {
     /// If no encoding is specified, defaults to UTF-8.
     #[inline]
     #[cfg(feature = "encoding")]
-    pub fn decode_without_bom<'b, 'c>(&'b self, bytes: &'c [u8]) -> Cow<'c, str> {
-        self.encoding.decode_with_bom_removal(bytes).0
+    pub fn decode_without_bom<'b, 'c>(&'b mut self, mut bytes: &'c [u8]) -> Cow<'c, str> {
+        if self.is_encoding_set {
+            return self.encoding.decode_with_bom_removal(bytes).0;
+        }
+        if bytes.starts_with(b"\xEF\xBB\xBF") {
+            self.is_encoding_set = true;
+            bytes = &bytes[3..];
+        } else if bytes.starts_with(b"\xFF\xFE") {
+            self.is_encoding_set = true;
+            self.encoding = UTF_16LE;
+            bytes = &bytes[2..];
+        } else if bytes.starts_with(b"\xFE\xFF") {
+            self.is_encoding_set = true;
+            self.encoding = UTF_16BE;
+            bytes = &bytes[3..];
```

Comment:
Here, bytes will be
```
[  254,    255,    0 ]
```
so I had to make ```&bytes[2..]``` to ```&bytes[3..]```.
(This is strange but I don't know why I had 0 in the end of slice.)